### PR TITLE
Adding optional prop for lazy load the intercom script

### PR DESCRIPTION
**Pull request by Siggnja**
Solves #237

### CODE REVIEW

**Comment by devrnt** [comment]
Thanks for the effort that you've put into this. 

I'll wait for your response on #237. But I'm not really eager to change this initialize snippet since it's copied from the official Intercom SPA manual. I will also have to take a look at the side effects of this `setInterval` even with a 0ms timeout it will put it on the event loop

**Comment by Siggnja** [comment]
@devrnt Yes I noticed the failing test, I didn't want to put in more effort until I got some feedback from you. If this is something you would be willing to add to the codebase I would of course make sure this is a progressive enhancement and that it won't affect current functionality.

I could even look into adding a test that covers this new functionality.

File: src/provider.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ export const IntercomProvider: React.FC<IntercomProviderProps> = ({
   onUnreadCountChange,
   shouldInitialize = !isSSR,
   apiBase,
+  initializeDelayInMs,
```
Comment (devrnt) [inline src/provider.tsx:25]: I would call this just `initializeDelay`, we'll mention the unit in the jsdocs

File: src/types.ts [hunk 2]
```diff
@@ -419,4 +419,10 @@ export type IntercomProviderProps = {
    * Format https://${INTERCOM_APP_ID}.intercom-messenger.com
    */
   apiBase?: string;
+  /**
+   * Indicates if the intercom initialization should be delayed
```
Comment (devrnt) [inline src/types.ts:423]: Mention that this is in ms

**Review by devrnt — CHANGES_REQUESTED** [review]

**Comment by devrnt** [comment]
@Siggnja Looking good, I would like to see a small section in the `README` that "explains" this functionality and also update the API table

**Comment by Siggnja** [comment]
@devrnt Thanks for the comments, I've got some time tomorrow to update the PR and maybe add some tests.

File: src/types.ts (hunk 2 above)
Comment (Siggnja) [reply src/types.ts]: Done

File: src/provider.tsx (hunk 1 above)
Comment (Siggnja) [reply src/provider.tsx]: Done in https://github.com/devrnt/react-use-intercom/pull/236/commits/2b733463f2a223babaf1e262063b291c1882bf6a

File: src/types.ts (hunk 2 above)
Comment (Siggnja) [reply src/types.ts]: Done in https://github.com/devrnt/react-use-intercom/pull/236/commits/2b733463f2a223babaf1e262063b291c1882bf6a

**Comment by Siggnja** [comment]
@devrnt I've added a test, let me know if you were thinking about something different.

One thing I thought about while implementing this was whether we should support loading the intercom script before the delay if the user clicks/hovers over a custom launcher before the timeout has ended. This is my particular use case, we've got our own launcher which triggers the .show() action when clicked. This would mean that if users click the launcher before the delay they wouldn't experience a relatively any input delay. This is of course a special case which users would rarely encounter.

File: playground/modules/useIntercom/useIntercomWithDelay.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const {
+    boot,
+  } = useIntercom();
+  const handleBoot = React.useCallback(() => boot(), [boot]);
+
+  return (
+    <Grid>
+      <Item>
+        <p>
+          Intercom will be autobooted after 5000ms
```
Comment (devrnt) [inline playground/modules/useIntercom/useIntercomWithDelay.tsx:36]: Awesome job that you included an example, thanks for this!

Just a small note I would mention here: Intercom will be initialized (and autobooted) after 5000ms

**Review by devrnt — COMMENTED** [review]
First of all, thank you for all the effort, really appreciate it!

> @devrnt I've added a test, let me know if you were thinking about something different.

I'll check these asap

> One thing I thought about while implementing this was whether we should support loading the intercom script before the delay if the user clicks/hovers over a custom launcher before the timeout has ended. This is my particular use case, we've got our own launcher which triggers the .show() action when clicked. This would mean that if users click the launcher before the delay they wouldn't experience a relatively any input delay. This is of course a special case which users would rarely encounter.

I've been thinking the same, however introducing a loading state seems too complex, especially since Intercom wants to avoid this themself. I declare myself: there is no way to check the "has initialized state" with the vanilla snippet, invoking methods before the snippet is initialized just doesn't do anything.

However, something that could be done is keeping track of the `isInitialized` state. Again maybe this introduces a bit too much complexity for users which is sufficient with a simple implementation. Also, it will probably mean that we'll have to re-render the whole thing, maybe a dealbreaker since at the moment it never rerenders

**Comment by devrnt** [comment]
Alright, the test seems good to me. I've added a section in the `README` about this initialization. 

Just one small change and we're good to go 🚀

**Comment by Siggnja** [comment]
@devrnt Addressed your comment and removed some unused stuff from the example. Please take a look and let me know if there is anything else you need from me.

**Review by devrnt — APPROVED** [review]

**Comment by devrnt** [comment]
Released in [v1.2.0](https://github.com/devrnt/react-use-intercom/releases/tag/v1.2.0)

**Pull request merged**